service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/CaseEventHandler.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.hmcts.reform.fpl.handlers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.events.CaseDataChanged;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.tasklist.Task;
+import uk.gov.hmcts.reform.fpl.service.TaskListRenderer;
+import uk.gov.hmcts.reform.fpl.service.TaskListService;
+import uk.gov.hmcts.reform.fpl.service.ccd.CoreCaseDataService;
+
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.CASE_TYPE;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.JURISDICTION;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDetailsHelper.isInOpenState;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class CaseEventHandler {
+
+    private final ObjectMapper mapper;
+    private final CoreCaseDataService coreCaseDataService;
+    private final TaskListService taskListService;
+    private final TaskListRenderer taskListRenderer;
+
+    @EventListener
+    public void handleCaseDataChange(final CaseDataChanged event) {
```

Comment:
Do we want to handle this asynchronusly or is the performance hit not that large? If we did do it asynchronusly would this then lead to some flakiness in the e2e tests or unpredictibility on tab updates for the end user?

Reply:
nope, if it is done asynchronously then task list wont be refreshed once user finish event

Reply:
Thought that might be the case.